reporter/src/main/kotlin/utils/SpdxDocumentModelMapper.kt

```diff
@@ -194,6 +197,22 @@ private fun getSpdxCopyrightText(
 
 private fun Identifier.toSpdxPackageName(): String = "$type:$namespace:$name"
 
+private fun Package.toSpdxExternalReferences(): List<SpdxExternalReference> {
+    val externalReferences = mutableListOf<SpdxExternalReference>()
+
+    if (purl.isNotBlank()) {
```

Comment:
A pURL can only ever be *empty*, so we should be more strict here. Also, I'd make this function return early in this case to reduce indentation, like `if (purl.isEmpty()) return null`.